spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryTests.java

```diff
@@ -330,6 +352,113 @@ public void shouldPullClean() throws Exception {
+		envRepository.setUri("git+ssh://git@somegitserver/somegitrepo");
+		envRepository.setBasedir(new File("./mybasedir"));
+		final String username = "someuser";
+		final String password = "mypassword";
+		envRepository.setUsername(username);
+		envRepository.setPassword(password);
+		envRepository.setCloneOnStart(true);
+		envRepository.afterPropertiesSet();
+
+		assertTrue(mockCloneCommand.getCredentialsProvider() instanceof UsernamePasswordCredentialsProvider);
+
+		CredentialsProvider provider = mockCloneCommand.getCredentialsProvider();
+		CredentialItem.Username usernameCredential = new CredentialItem.Username();
+		CredentialItem.Password passwordCredential = new CredentialItem.Password();
+		assertTrue(provider.supports(usernameCredential));
+		assertTrue(provider.supports(passwordCredential));
+
+		provider.get(new URIish(), usernameCredential);
+		assertEquals(usernameCredential.getValue(), username);
+		provider.get(new URIish(), passwordCredential);
+		assertEquals(String.valueOf(passwordCredential.getValue()), password);
+	}
+
+	@Test
+	public void passphraseShouldSetCredentials() throws Exception {
+		final String passphrase = "mypassphrase";
+		Git mockGit = mock(Git.class);
+		MockCloneCommand mockCloneCommand = new MockCloneCommand(mockGit);
+
+		JGitEnvironmentRepository envRepository = new JGitEnvironmentRepository(this.environment);
+		envRepository.setGitFactory(new MockGitFactory(mockGit, mockCloneCommand));
+		envRepository.setUri("git+ssh://git@somegitserver/somegitrepo");
+		envRepository.setBasedir(new File("./mybasedir"));
+		envRepository.setPassphrase(passphrase);
+		envRepository.setCloneOnStart(true);
+		envRepository.afterPropertiesSet();
+
+		assertTrue(mockCloneCommand.hasPassphraseCredentialsProvider());
+
+		CredentialsProvider provider = mockCloneCommand.getCredentialsProvider();
+		assertFalse(provider.isInteractive());
+
+		CredentialItem.StringType stringCredential = new CredentialItem.StringType(PassphraseCredentialsProvider.PROMPT, true);
+
+		assertTrue(provider.supports(stringCredential));
+		provider.get(new URIish(), stringCredential);
+		assertEquals(stringCredential.getValue(), passphrase);
+	}
+
+	@Test
+	public void strictHostKeyCheckShouldCheck() throws Exception {
+		String uri = "git+ssh://git@somegitserver/somegitrepo";
+		SshSessionFactory.setInstance(null);
+		JGitEnvironmentRepository envRepository = new JGitEnvironmentRepository(this.environment);
+		envRepository.setUri(uri);
+		envRepository.setBasedir(new File("./mybasedir"));
+		envRepository.setStrictHostKeyChecking(true);
+		envRepository.setCloneOnStart(true);
+		try {
+			// this will throw but we don't care about connecting.
+			envRepository.afterPropertiesSet();
+		} catch (Exception e) {
+			final OpenSshConfig.Host hc = OpenSshConfig.get(FS.detect()).lookup("github.com");
+			JschConfigSessionFactory factory = (JschConfigSessionFactory) SshSessionFactory.getInstance();
```

Comment:
This test is a little hairy but I couldn't figure out a way to test that the config was being respected in jGit. If you'd like me to approach this test differently please let me know.